client/src/components/Header/Header.test.js

```diff
@@ -46,9 +47,15 @@ describe('<NavLinks />', () => {
       },
       toggleNightMode: theme => theme
     };
+
+    const testComp = React.createElement(
```

Comment:
If someone could help me figure out how to test with translations

Reply:
Hey Shaun, what issue are you having?

I wouldn't say I'm an expert in this, but I have written and reviewed a couple tests that involved translation (or these react-i18next stuff), so I think I help take a look.

Reply:
Hey @huyenltnguyen any help would be much appreciated.

For these tests, I want to ensure something like the `to` prop for one of the `Nav` children is `https://forum.freecodecamp.org/`. However, it does not give me the compiled code - it gives me `links:nav.forum`.

Reply:
Ah yeah. This sadly is what has been bugging me. I had to resorted to the translation token in the tests instead of looking for the real string.

I'll see if I can find a way to make it spit out the actual string.

Reply:
Well, one solution for this case is that we don't use translation strings for the links.

So instead of 

```tsx
const NavLinks = () => {
  return (
    <Link to={t('links:nav.forum')} />
  )
}
```

we could have the links mapped before passing them to `Link`

```tsx
const getForumLink = (lang) => {
  switch (lang) {
	case 'en':
      return ...
    case 'es':
      return ...
  } 
}

const NavLinks = () => {
  const { i18n } = useTranslation();

  const forumLink = getForumLink(i18n.language) 

  return (
    <Link to={forumLink} />
  )
}
```

This doesn't resolve the original problem, but it could at least unblock you for now. But to address the token issue, the idea I'm having right now is to save English strings as translation keys. Say if we have a string 'Hello world', our JSON would be

```jsonc
// english/translations.json
{
  'Hello world': 'Hello world'
}

// german/translations.json
{
  'Hello world': 'Hallo Welt'
}
```

This would allow us to see the expected string in our tests (because the key and the value are identical), but this applies to English only, so it is helpful in tests that are for business logic where we just want to know if a specific string is rendered, but it won't work for tests that require actual translated strings. 

This approach is also quite unrealistic, as it impacts the i18n framework that we already have in place, so

Reply:
@huyenltnguyen How is that different to what we have:
```json
"nav": {
    "forum": "https://forum.freecodecamp.org/",
    "news": "https://freecodecamp.org/news/"
  }
```

Reply:
I'm not sure if you're asking about replacing the real strings or saving the token in a different way.

If it's the latter, then our JSON would be

```json
"nav": {
  "https://forum.freecodecamp.org/": "https://forum.freecodecamp.org/",
  "https://freecodecamp.org/news/": "https://freecodecamp.org/news/"
}
```

It doesn't change the way we write tests, but at least we know what the expected string should be

```js
expect(link).toEqual('links:nav.forum') 

// vs

expect(link).toEqual('links:nav['https://forum.freecodecamp.org/']')
```

Again, this doesn't resolve the original issue, as we are working around here by assigning the translation value to the key, and this approach doesn't work in tests that assert if the translated string is correctly applied. This is the only solution I have at the moment though. I'll need to look into it a bit more.

---
But regarding the approach, I'm not really sure why we decided to switch to translation tokens instead of computing and using the strings directly. My thought is that:
- These are constants and are not translatable, so they are not needed to be exposed to translators (unless I'm overlooking)
- Translators could accidentally change the links, now that they are editable. This can totally be caught when we review translation PRs, but it adds more things for us to look for, and we could just miss them (because we are human)
- IMHO, having the links centralized would be more convenient when we need to look them up or when we want to update them, instead of navigating through the ever growing translation files

That's just my 2 cents 🙂

Reply:
> Translators could accidentally change the links, now that they are editable.

FWIW, the `links.json` files don't go through Crowdin.

Reply:
Ah, I didn't know that, so this change is totally safe then.

Thanks Nick!